tabular/src/autogluon/tabular/models/xgboost/callbacks.py

```diff
@@ -0,0 +1,193 @@
+import time
+import psutil
+import logging
+from xgboost.core import EarlyStopException
+
+logger = logging.getLogger(__name__)
+
+
+def _fmt_metric(value, show_stdv=True):
+    """format metric string"""
+    if len(value) == 2:
+        return '{0}:{1:.5f}'.format(value[0], value[1])
+    if len(value) == 3:
+        if show_stdv:
+            return  '{0}:{1:.5f}+{2:.5f}'.format(value[0], value[1], value[2])
+        return '{0}:{1:.5f}'.format(value[0], value[1])
+    raise ValueError("wrong metric value")
+
+
+def print_evaluation(period=1, show_stdv=True):
+    """Create a callback that print evaluation result.
+    We print the evaluation results every **period** iterations
+    and on the first and the last iterations.
+    Parameters
+    ----------
+    period : int
+        The period to log the evaluation results
+    show_stdv : bool, optional
+         Whether show stdv if provided
+    Returns
+    -------
+    callback : function
+        A callback that print evaluation every period iterations.
+    """
+    def callback(env):
+        """internal function"""
+        if env.rank != 0 or (not env.evaluation_result_list) or period is False or period == 0:
+            return
+        i = env.iteration
+        if i % period == 0 or i + 1 == env.begin_iteration or i + 1 == env.end_iteration:
+            msg = '\t'.join([_fmt_metric(x, show_stdv) for x in env.evaluation_result_list])
+            logger.log(20, '[%d]\t%s\n' % (i, msg))
```

Comment:
No need for `\n` in the log messages, as logger.log already adds a `\n` to every message. This applies to every log message in `early_stop_custom` as well.

Reply:
I will remove all `\n` at the end log message.